src/solvers/applyexp.jl

```diff
@@ -0,0 +1,103 @@
+using Printf: @printf
+import ConstructionBase: setproperties
+
+@kwdef mutable struct ApplyExpProblem{State}
+  state::State
+  operator
+  current_time::Number = 0.0
+end
```

Comment:
Related to my comment above about `SweepIterator`, I think we should parameterize the problem types. Also related to my other comment I think we should switch the argument ordering so the `operator` is first.

Also, I'm suspicious when I see `0.0` since that is a `Float64`. If the time is `Float64` it might then promote the element type of other objects it interacts with to double precision, which would be bad on GPU where single precision is better. We should have a constructor that determines the number type from the inputs, such as: `ApplyExpProblem(operator, state) = ApplyExpProblem(operator, state, zero(promote_type(scalartype(operator), scalartype(state)))`.